src/components/flex-container/flex-container.scss

```diff
@@ -49,3 +46,17 @@
 :host(limel-flex-container[justify='space-evenly']) {
     justify-content: space-evenly;
 }
+
+// Since consumers often copy/paste from examples of `limel-dialog` from the
+// documentations, they often put their buttons in a `limel-flex-container`
+// which is used in the footer slot of those examples with buttons.
+// So to prevent unwanted layout problems of buttons having no distance, we use
+// below code to target `limel-flex-container`s in those slots.
+:host(limel-flex-container[slot='button']) {
+    gap: 0.5rem; // Makes sure buttons get some default distance
+    width: 100%;
+    justify-content: flex-end;
+}
+:host(limel-flex-container[slot='button'][direction='horizontal'][reverse]) {
+    justify-content: flex-start;
```

Comment:
I added this so that we don't "break" existing solutions that use `reverse={true}`.

Now:
![image](https://user-images.githubusercontent.com/154725/124102842-fd850180-da60-11eb-8bf6-6ae47f16d86b.png)

Before:
![image](https://user-images.githubusercontent.com/154725/124102794-f52cc680-da60-11eb-9616-4d1165c8878d.png)

Reply:
smart